### ISSUE TITLE
False positives on jsx-no-useless-fragment when the fragment contains a function call that returns an array

### ISSUE THREAD

**Issue by suchanlee** [yannickcr/eslint-plugin-react#2512]
There are false positives on jsx-no-useless-fragment when the fragment contains a function call that returns an array.

Sample code:
```
    function getDivs() {
      return [<div />, <div />];
    }

    function component() {
      return <React.Fragment>{getDivs()}</React.Fragment>;
    }
```

In the above code the linter will error with the message:
`Fragments should contain more than one child - otherwise, there‘s no need for a Fragment at all.`

**Comment by ljharb** [comment]
Short of you disabling the rule in this one case, what would you suggest? There's no way to statically know what kind of thing a function call returns.

**Comment by suchanlee** [comment]
ah that's true, I was head deep in writing eslint rules with the typescript parser which does allow statical type analysis so had just assumed that it would be available here as well. Closing.

**Comment by nicu-chiciuc** [comment]
@ljharb Would it be possible to statically figure out if a function returns an array if this rule was based on `typescript-eslint`? In that case it would have access to type information.

**Comment by ljharb** [comment]
Yes, theoretically, but then it wouldn’t provide any benefit for JS users.
